Added a physics system that has a gravity feature

Ranko030 opened the pull request.

gavrilovmiroslav committed:
Created full FSM implementation and implemented a platformer example using this system

gavrilovmiroslav committed:
API streamlined a bit more

gavrilovmiroslav committed:
Update character_controller_fsm.cpp

gavrilovmiroslav committed:
Update character_controller_fsm.cpp

UbiBelETF committed:
Merge pull request #92 from UbiBelETF/engine/fsms

Created full FSM implementation and implemented a platformer example …

DPrivalov committed:
fixed build errors in an easy way

DPrivalov committed:
Merge pull request #96 from UbiBelETF/engine/mac_fixes

fixed build errors in an easy way

Ranko030 committed:
started developing physics so that the game hase some kind of gravity

Ranko030 commented:
Also added an ability that a hero can have a double jump.

gavrilovmiroslav committed:
Mac fix unfixed windows, windows fix applied

gavrilovmiroslav committed:
Fixed depth scaling by increasing depth by two orders of magnitude. It will still happen, but only at unimaginable scales. Needs testing. Fixes #94

UbiBelETF committed:
Merge pull request #100 from UbiBelETF/engine/inputs-crash

Mac fix unfixed windows, windows fix applied

gavrilovmiroslav committed:
Styleguide issue: added E to start of CharacterStates

Commit:
Added simple aiming system

Commit:
Added comment for setting up rotate command

gavrilovmiroslav committed:
Merge pull request #101 from UbiBelETF/engine/camera-depth-scale-fix

Engine/camera depth scale fix

Commit:
Comment refactored

Coakim committed:
Merge pull request #105 from UbiBelETF/feature/AimingSystem

Added simple aiming system

Commit:
Fix for Aiming System feature

Coakim committed:
Merge pull request #108 from UbiBelETF/fix/AimingSystem

Fix for Aiming System feature

DPrivalov committed:
opengl framebuffer size fix

UbiBelETF committed:
Merge pull request #113 from UbiBelETF/engine/mac-opengl-fix

opengl framebuffer size fix

Ranko030 committed:
improved physics but removed player controls(not all).Player can now only run left and right

Ranko030 committed:
fixed minor mistake

UbiBelETF commented on source/dagger/gameplay/team_game/brawler_controller.cpp:
```diff
@@ -27,88 +30,73 @@ void BrawlerControllerSystem::SpinUp()
+                physics_.velocity.x += sprite_.scale.x* char_.speed;
+                char_.run = true;
+            }
+            /*if(!char_.jump )
```
Why keep such a large comment? It's confusing to review, this way.

Ranko030 replied:
Oh sorry forgot to delete that.

Ranko030 committed:
removed the comments

Ranko030 committed:
Added an invisible wall to the left and when you run on it , it collides but then the game freezes

gavrilovmiroslav committed:
Made a debug fix to help solve the crashing issue

Ranko030 committed:
fixed the collisions

Ranko030 committed:
Merge pull request #117 from UbiBelETF/team/tired_bunch/test/gravity

Fixed the wall collide

Ranko030 committed:
fixed conflicts

DPrivalov commented:
Check SimpleCollisions.cpp  file, SimpleCollisionsSystem::Run should reset collision.colided status before starting to process new frame.

Ranko030 committed:
Merge branch 'team/tired_bunch/feature/physics_beginning' of https://github.com/UBIBelETF/dagger into team/tired_bunch/feature/physics_beginning

Ranko030 commented:
> Check SimpleCollisions.cpp file, SimpleCollisionsSystem::Run should reset collision.colided status before starting to process new frame.

Well, I thought of removing the SimpleCollisionsSystem and just using the SimpleCollision struct in CollisionSystem to see if there will be a collision in the next frame. Now if there will be a collision then just try to lower the velocity to the point when there won't be one.

DPrivalov commented:
@Ranko030 , cool
mention me in comments when you implement it.

Ranko030 commented:
@DPrivalov check the collisions.cpp file. In there I implemented that system . That is the thing I added from the team/tired_bunch/test/gravity branch when I fixed it.

Ranko030 committed:
Added the FSM (controlls and animations) for the player

Ranko030 committed:
removed unnecssary comments

Ranko030 committed:
fixed the FSM

Ranko030 committed:
fixed the edges on gravity-collision system
